src/plugins/intel_gpu/src/graph/graph_optimizer/reorder_inputs.cpp

```diff
@@ -565,6 +602,33 @@ void reorder_inputs::run(program& p, layout_optimizer& lo, reorder_factory& rf)
     };
 
     const auto reorder_convolution = [&p, &lo, &rf, &debug_config](typed_program_node<convolution>& conv_node) {
+        {
+            // Change input data type of conv node from i32 to f32
+            auto& input = conv_node.input();
+            auto input_layout = input.get_output_layout();
+            if (input_layout.data_type == data_types::i32) {
+                auto new_layout = input_layout;
+                new_layout.data_type = data_types::f32;
+                auto new_input = rf.get_reorder(input.id(), input_layout, new_layout);
+                if (new_input.first) {
+                    p.add_intermediate(new_input.first, conv_node, 0, !new_input.second);
+                }
+            }
+
+            // Change weights type i32 to f32
+            auto& weights = conv_node.weights();
+            auto weights_layout = weights.get_output_layout();
+            if (weights_layout.data_type == data_types::i32) {
+                auto new_layout = weights_layout;
+                new_layout.data_type = data_types::f32;
+                auto new_input = rf.get_reorder(weights.id(), weights_layout, new_layout);
+                if (new_input.first) {
+                    p.add_intermediate(new_input.first, conv_node, 1, !new_input.second);
+                    p.get_or_create(new_input.first).recalc_output_layout(false);
+                }
+            }
+        }
```

Comment:
There is some reorder for handling weights below this code, and it is checking whether the weight input is "data" or "constant" .  I am afraid that adding this here will affect those condition. How do you think? I thinkd we'd better change the location of this code not to touch the behavior of the below code.  (e.g., move this after the below condition ?)

Reply:
done